Project.toml

```diff
@@ -42,6 +42,8 @@ TimerOutputs = "a759f4b9-e2f1-59dc-863e-4aeb61b1ea8f"
 Triangulate = "f7e6ffb2-c36d-4f8f-a77e-16e897189344"
 TriplotBase = "981d1d27-644d-49a2-9326-4793e63143c3"
 TriplotRecipes = "808ab39a-a642-4abf-81ff-4cb34ebbffa3"
+DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
+OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
```

Comment:
It is also possible to avoid adding the dependency `OrdinaryDiffEq` by copying the routine `modify_dt_for_tstops!` into Trixi.jl.

Reply:
OrdinaryDiffEq.jl is a huge beast (initial latency, precompile times, ...). Is it possible to avoid this or use an approach similar to DiffEqCallbacks.jl?
Copying code and then having to maintain it is maybe also not really great...

Reply:
Given that the last non-formatting-related change to the function was ~3 years ago (https://github.com/SciML/OrdinaryDiffEq.jl/blame/8a0329bfcc40c39a3ff4ef6d69e195fc4d784e4e/src/integrators/integrator_utils.jl#L33-L47), I would find it acceptable to copy the function in this specific case, if no other way can be found.

Reply:
Thanks @ranocha and @sloede. For the time being, I removed the dependency in d04fd98 by copying this routine. Please let me know if another approach is preferred. 

> Is it possible to avoid this or use an approach similar to DiffEqCallbacks.jl?

What is exactly this approach? I see that DiffEqCallbacks.jl is part of the dependencies with the `[compat]` entry `DiffEqCallbacks = "2.25"`.

Reply:
> > Is it possible to avoid this or use an approach similar to DiffEqCallbacks.jl?
> 
> What is exactly this approach?

IIRC, DiffEqCallbacks.jl was created as a spin-off to have the ability to implement callbacks with a (somewhat) lightweight dependency, i.e., without having to pull in the entire OrdinaryDiffEq.jl/DifferentialEquations.jl beast. IIUC, the question refers to whether this would be possible here as well, i.e., to include a smaller, lightweight package from which the desired functionality can be imported.

Reply:
Thanks for the explanation. Unfortunately, DiffEqCallbacks.jl does not contain the function `modify_dt_for_tstops!`, and I don't see any other lightweight package where it is included. Therefore, if there are no other suggestions, for the moment I'll stick to using the function [modify_dt_for_tstops!(integrator::SimpleIntegratorSSP)](https://github.com/trixi-framework/Trixi.jl/blob/4322f5b53ec6b56c96114405a8f82553d8b335e5/src/time_integration/methods_SSP.jl#L270), which is a copy of OrdinaryDiffEq's function into methods_SSP.jl.

Reply:
What I meant was to use a package extension loading the code from OrdinaryDiffEq.jl when necessary.

Reply:
Copying is easier - we can just keep the current version